diff --git a/config.php b/config.php
index b3d9bbc..1ef312e 100644
--- a/config.php
+++ b/config.php
@@ -1 +1,8 @@
 <?php
+
+$database_config = [
+    'host' => 'yeticave-mysql',
+    'user' => 'root',
+    'password' => 'password',
+    'database' => 'yeticave',
+];
diff --git a/functions.php b/functions.php
index 5e1bc30..df936b8 100644
--- a/functions.php
+++ b/functions.php
@@ -1,5 +1,7 @@
 <?php
 
+require_once 'init.php';
+
 function include_template($name, $data)
 {
     $name = 'templates/' . $name;
@@ -89,7 +91,7 @@ function get_lots($link): array
 }
 
 
-function get_lot_single($link, $lot_id)
+function get_lot_single($link, int $lot_id): array
 {
     $sql_lot_single = "SELECT l.id, l.created_at, l.title, l.description, l.img_url, l.price_start, l.end_at, l.bet_step, l.category_id, l.user_id, l.winner_user_id, c.title as cat_title FROM lot l
       JOIN category c
@@ -102,29 +104,31 @@ function get_lot_single($link, $lot_id)
         die(include_template('error.php', ['error' => $error]));
     }
 
-    //echo $result_lot_single->num_rows;
-    if ($result_lot_single->num_rows == 0) {
-
-        $error = include_template(
-            '404.php',
-            [
-                'categories' => $categories,
-                'error' => 'Ошибка 404! Страница не найдена'
-            ]
-        );
-
-        die(include_template('layout.php',
-            [
-                'content' => $error,
-                'site_title' => $site_title,
-                'user_name' => $user_name,
-                'user_avatar' => $user_avatar,
-                'is_auth' => $is_auth,
-                'categories' => $categories,
-            ]
-        ));
+    return mysqli_fetch_assoc($result_lot_single) ?? [];
+}
+
+function get_layout(string $content, array $categories): string {
 
-    }
+    $init_data = get_init_data();
 
-    return mysqli_fetch_assoc($result_lot_single);
+    return include_template(
+        'layout.php',
+        array_merge($init_data, [
+            'content' => $content,
+            'categories' => $categories,
+        ])
+    );
+}
+
+function get_layout_404(array $categories): string
+{
+    $error = include_template(
+        '404.php',
+        [
+            'categories' => $categories,
+            'error' => 'Ошибка 404! Страница не найдена'
+        ]
+    );
+
+    return get_layout($error, $categories);
 }
diff --git a/index.php b/index.php
index ec70ae5..f66ca75 100644
--- a/index.php
+++ b/index.php
@@ -1,22 +1,6 @@
 ﻿<?php
-$is_auth = rand(0, 1);
 
-$user_name = 'Kirill';
-$user_avatar = 'img/user.jpg';
-
-$site_title = 'YetiCave - интернет-аукцион';
-
-
-require_once 'functions.php';
-
-$database_config = [
-    'host' => 'localhost',
-    'user' => 'root',
-    'password' => '',
-    'database' => 'yeticave10'
-];
-
-$link = init_database($database_config);
+require_once 'init.php';
 
 $categories = get_categories($link);
 $lots = get_lots($link);
@@ -30,16 +14,4 @@ $main = include_template(
     ]
 );
 
-$layout = include_template(
-    'layout.php',
-    [
-        'content' => $main,
-        'site_title' => $site_title,
-        'user_name' => $user_name,
-        'user_avatar' => $user_avatar,
-        'is_auth' => $is_auth,
-        'categories' => $categories,
-    ]
-);
-
-print($layout);
+print(get_layout($main, $categories));
diff --git a/init.php b/init.php
index b3d9bbc..4f20d6c 100644
--- a/init.php
+++ b/init.php
@@ -1 +1,15 @@
 <?php
+
+require_once 'functions.php';
+require_once 'config.php';
+
+function get_init_data(): array {
+    return [
+        'is_auth' => rand(0, 1),
+        'user_name' => 'Kirill',
+        'user_avatar' => 'img/user.jpg',
+        'site_title' => 'YetiCave - интернет-аукцион',
+    ];
+}
+
+$link = init_database($database_config);
diff --git a/lot.php b/lot.php
index 22a8dbb..87ee29e 100644
--- a/lot.php
+++ b/lot.php
@@ -1,54 +1,24 @@
-<?php // 2. Начало сценария-)
-$is_auth = rand(0, 1);
+<?php
 
-$user_name = 'Kirill';
-$user_avatar = 'img/user.jpg';
-
-$site_title = 'YetiCave - интернет-аукцион';
-
-
-require_once 'functions.php';
-
-$database_config = [
-    'host' => 'localhost',
-    'user' => 'root',
-    'password' => '',
-    'database' => 'yeticave10'
-];
-
-$link = init_database($database_config);
+require_once 'init.php';
 
 $categories = get_categories($link);
 
 
-if (isset($_GET['id'])) {
-    $lot_id = $_GET['id'];
+if (!isset($_GET['id'])) {
+    die(get_layout_404($categories));
 }
-else {
-    $error = include_template(
-        '404.php',
-        [
-            'categories' => $categories,
-            'error' => 'Ошибка 404! Страница не найдена'
-        ]
-    );
 
-    die(include_template('layout.php',
-        [
-            'content' => $error,
-            'site_title' => $site_title,
-            'user_name' => $user_name,
-            'user_avatar' => $user_avatar,
-            'is_auth' => $is_auth,
-            'categories' => $categories,
-        ]
-    ));
-}
+$lot_id = (int) $_GET['id'];
 
-$lot_single = get_lot_single($link, $_GET['id']);
+$lot_single = get_lot_single($link, $lot_id);
 
+if (empty($lot_single)) {
+    die(get_layout_404($categories));
+}
 
-$lot = include_template(
+
+$lot_page = include_template(
     'lot.php',
     [
         'categories' => $categories,
@@ -56,16 +26,4 @@ $lot = include_template(
     ]
 );
 
-$layout = include_template(
-    'layout.php',
-    [
-        'content' => $lot,
-        'site_title' => $site_title,
-        'user_name' => $user_name,
-        'user_avatar' => $user_avatar,
-        'is_auth' => $is_auth,
-        'categories' => $categories,
-    ]
-);
-
-print($layout);
+print(get_layout($lot_page, $categories));
diff --git a/templates/404.php b/templates/404.php
index 7bf380e..ecf1ce7 100644
--- a/templates/404.php
+++ b/templates/404.php
@@ -1,19 +1,7 @@
-    <main class="404">
-        <nav class="nav">
-            <ul class="nav__list container">
-
-                <?php foreach ($categories as $category): ?>
-                    <li class="nav__item">
-                        <a href="pages/all-lots.html"><?= h($category['title']); ?></a>
-                    </li>
-                <?php endforeach; ?>
-
-            </ul>
-        </nav>
-
-        <section class="lot-item container">
-            <h2>404 Страница не найдена</h2>
-            <p>Данной страницы не существует на сайте.</p>
-        </section>
-    </main>
+<main class="404">
+    <section class="lot-item container">
+        <h2>404 Страница не найдена</h2>
+        <p>Данной страницы не существует на сайте.</p>
+    </section>
+</main>
 
diff --git a/templates/layout.php b/templates/layout.php
index e067986..56ac285 100644
--- a/templates/layout.php
+++ b/templates/layout.php
@@ -46,6 +46,8 @@
     </header>
 
 
+
+
     <?= $content; ?>
 
 </div>
